x/pairing/keeper/unresponsive_provider.go

```diff
@@ -41,68 +42,65 @@ func (k Keeper) UnstakeUnresponsiveProviders(ctx sdk.Context, epochsNumToCheckCU
 	return nil
 }
 
-// Function that returns a map that links between a provider that should be punished and its punishEntry
+// Function that returns a map that links between a provider that should be punished and its providerCuCounterForUnreponsiveness
 func (k Keeper) getUnresponsiveProvidersToPunish(ctx sdk.Context, currentEpoch uint64, epochsNumToCheckCUForUnresponsiveProvider uint64, epochsNumToCheckCUForComplainers uint64) (map[string][]string, error) {
 	// check the epochsNum consts
 	if epochsNumToCheckCUForComplainers <= 0 || epochsNumToCheckCUForUnresponsiveProvider <= 0 {
 		return nil, utils.LavaError(ctx, k.Logger(ctx), "get_unresponsive_providers_to_punish", nil, "epochsNumToCheckCUForUnresponsiveProvider or epochsNumToCheckCUForComplainers are smaller or equal than zero")
 	}
 
-	providersPunishEntryMap := make(map[string]punishEntry) // map of punishEntryKey keys (string) with punishEntry values (punishEntryKey = providerAddress_chainID)
-	providersToPunishMap := make(map[string][]string)       // map of punishEntryKey keys (string) with list of providerPaymentStorage indices values (string array) (punishEntryKey = providerAddress_chainID)
-
 	// Get recommendedEpochNumToCollectPayment
 	recommendedEpochNumToCollectPayment := k.RecommendedEpochNumToCollectPayment(ctx)
 
-	// handle very early epochs (basically, don't punish)
-	epochBlocks, err := k.epochStorageKeeper.EpochBlocks(ctx, currentEpoch)
-	if err != nil {
-		return nil, utils.LavaError(ctx, k.Logger(ctx), "get_epoch_blocks", map[string]string{"err": err.Error(), "epoch": fmt.Sprintf("%+v", currentEpoch)}, "couldn't get epoch blocks param")
+	// To check for punishment, we have to go back (epochsNumToCheckCUForComplainers+epochsNumToCheckCUForUnresponsiveProvider) epochs from the current epoch. If there isn't enough memory, do nothing
+	epochTemp := currentEpoch
+	for counter := uint64(0); counter < epochsNumToCheckCUForComplainers+epochsNumToCheckCUForUnresponsiveProvider+recommendedEpochNumToCollectPayment; counter++ {
```

Comment:
why are we adding all 3 of those, should be something like 2+max(2,4) = 6 which is the max epoch to go back.
you did 2+2+4 = 8 which is far back

Reply:
done